### PR TITLE
Feature/gauge chart + cypress test cases

### PR DESCRIPTION
**Pull request by ramneet-persistent**
Gauge chart added
Gauge cypress test cases added

### CODE REVIEW

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx [hunk 1]
```diff
@@ -122,8 +122,9 @@ export const ConfigPanel = ({ visualizations, setCurVisId, callback, changeIsVal
     const isValid_valueOptions: { [key: string]: boolean } = {
       tree_map: curVisId === TreeMap && valueOptions?.childField?.length !== 0 &&
         valueOptions?.valueField?.length !== 0,
-      gauge: Boolean(curVisId === Gauge && valueOptions?.series && valueOptions.series?.length !== 0 &&
-        valueOptions?.value && valueOptions.value?.length !== 0),
+      // gauge: Boolean(curVisId === Gauge && valueOptions?.series && valueOptions.series?.length !== 0 &&
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx]: remove commented code.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_chart_options.tsx [hunk 2]
```diff
@@ -88,10 +88,32 @@ export const ConfigChartOptions = ({
...
+              ...btn,
+              label: btn.name,
+            })),
+            idSelected: vizState[schema.mapTo] || schema?.props?.defaultSelections[0]?.id,
+            handleButtonChange: handleConfigurationChange(schema.mapTo),
+            vizState,
+            ...schema.props,
+          };
         } else {
           params = {
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_chart_options.tsx:114]: check if param has added default state.

File: dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx [hunk 3]
```diff
@@ -61,45 +83,111 @@ export const Gauge = ({ visualizations, layout, config }: any) => {
...
           gauge: {
-            ...(thresholds && {
-              threshold: {
-                line: { color: thresholds[0]?.color || 'red', width: 4 },
-                thickness: 0.75,
-                value: thresholds[0]?.value || 0,
-              },
-            }),
+            // ...(showThresholdMarkers &&
+            //   thresholds &&
```
Comment (abasatwar) [inline dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx]: remove commented code.

File: dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx [hunk 4]
```diff
@@ -61,45 +83,111 @@ export const Gauge = ({ visualizations, layout, config }: any) => {
...
+            //     },
+            //   }),
+
+            //threshold labels
+            ...(showThresholdLabels && thresholds && thresholds.length
+              ? {
+                  axis: {
+                    ticktext: [gauge.value, ...thresholds.map((t: ThresholdUnitType) => t.name)],
+                    tickvals: [gauge.value, ...thresholds.map((t: ThresholdUnitType) => t.value)],
+                    ticklen: 5,
```
Comment (abasatwar) [inline dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx]: add 5 in constants

File: dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx [hunk 5]
```diff
@@ -61,45 +83,111 @@ export const Gauge = ({ visualizations, layout, config }: any) => {
...
+                }
+              : {}),
+            // multiple threshold markers!!!!
+            ...(showThresholdMarkers &&
+              thresholds &&
+              thresholds.length && {
+                steps: thresholds.map((threshold: ThresholdUnitType) => {
+                  const value = Number(threshold.value);
+                  return {
+                    range: [value, value + 0.25] /*width needs improvement*/,
```
Comment (abasatwar) [inline dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx]: add magic number in constants

File: dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx [hunk 6]
```diff
@@ -61,45 +83,111 @@ export const Gauge = ({ visualizations, layout, config }: any) => {
...
+    dataConfig?.panelOptions?.title,
+    orientation,
+  ]);
 
   const mergedConfigs = {
     ...config,
     ...(layoutConfig.config && layoutConfig.config),
   };
-
+
```
Comment (abasatwar) [inline dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx]: remove extra space added and prettify file.

**Review by abasatwar — CHANGES_REQUESTED** [review]

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx (hunk 1 above)
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx]: See if the commented code is required in the near future. If not, then remove commented code.

File: dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx [hunk 7]
```diff
@@ -61,45 +83,111 @@ export const Gauge = ({ visualizations, layout, config }: any) => {
...
+    valueSize,
+  ]);
 
   const mergedLayout = useMemo(() => {
     const isAtleastOneFullRow = Math.floor(gaugeData.length / PLOTLY_GAUGE_COLUMN_NUMBER) > 0;
     return {
       grid: {
-        rows: Math.floor(gaugeData.length / PLOTLY_GAUGE_COLUMN_NUMBER) + 1,
-        columns: isAtleastOneFullRow ? PLOTLY_GAUGE_COLUMN_NUMBER : gaugeData.length,
+        ...(orientation === 'auto' || orientation === 'h'
```
Comment (spattnaik) [inline dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx:159]: In future, try to create an enum for orientation and source the values from there. This is not a task for this PR. I also see the earlier code has been using 'h' and 'v' kind of values.

**Review by spattnaik — CHANGES_REQUESTED** [review]

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx (hunk 1 above)
Comment (ramneet-persistent) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx]: okay

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_chart_options.tsx (hunk 2 above)
Comment (ramneet-persistent) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/config_chart_options.tsx:114]: default value defined and add in all cases

File: dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx (hunk 3 above)
Comment (ramneet-persistent) [reply dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx]: done

Comment (ramneet-persistent) [reply dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx]: this was not actually finalized yet. as we haven't received confirmation on threshold @abasatwar

File: dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx (hunk 4 above)
Comment (ramneet-persistent) [reply dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx]: done

File: dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx (hunk 6 above)
Comment (ramneet-persistent) [reply dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx]: done

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx (hunk 1 above)
Comment (ramneet-persistent) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panel.tsx]: as series/values are moved to data config, this validation restricts  preview other chart options
so I added "true" value as valid series and values check 
commented code is removed now
@spattnaik

File: dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx (hunk 7 above)
Comment (ramneet-persistent) [reply dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx:159]: okay will consider enums in next module

File: dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx (hunk 5 above)
Comment (ramneet-persistent) [reply dashboards-observability/public/components/visualizations/charts/financial/gauge/gauge.tsx]: done

**Review by abasatwar — APPROVED** [review]

**Review by abasatwar — APPROVED** [review]

**Pull request merged**